public/locations-list-map.js

```diff
@@ -399,86 +465,147 @@ function initMap(states) {
 
      let $mapStats = $('#map-stats');
      if (singleStateFilter) {
-       // Center the map to a state if only one is set in query params
-       centerMapToState(map, firstState);
+          // Center the map to a state if only one is set in query params.
+          // Also, allow function to reset the initial coords/zoom globals.
+          centerMapToState(firstState, true);
 
-       // Update map stats (w/ selected state)
-       updateStats($mapStats, markers.length, [firstState]);
+          // Update map stats (w/ selected state)
+          updateStats($mapStats, markers.length, [firstState]);
 
      } else {
-       // Center the map on the nearest markers to the user if possible
-       centerMapToNearestMarkers(map, markers);
+          // Center the map on the nearest markers to the user if possible
+          centerMapToMarkersNearUser();
 
-       // Update stats (no states).
-       updateStats($mapStats, markers.length);
+          // Update stats (no states).
+          updateStats($mapStats, markers.length);
      }
+
+
+     // Initialize the map search autocompleter.
+     autocomplete = new google.maps.places.Autocomplete(
+       document.getElementById('map-search'), {types: ['geocode']}
+     );
+
+     // Avoid paying for data that you don't need by restricting the set of
+     // place fields that are returned to just the address components.
+     autocomplete.setFields(['geometry']);
+
+     // When the user selects an address from the drop-down, populate the
+     // address fields in the form.
+     autocomplete.addListener('place_changed', loadSearchInMap);
+
+     // Setup event listeners for map action links.
+     $('#use-location').on('click', function (e) {
+       e.preventDefault();
+       centerMapToMarkersNearUser();
+     });
+
+     $('#reset-map').on('click', function (e) {
```

Comment:
nit: use fat arrow function just to ensure consistent this semantics? Here and above?

Reply:
Force of habit, but will look later!

Reply:
going fix THESE instances that I created, but ignoring all the others that I didn't create b/c there are several others and not the right branch for this.